diff -Nur a/Makefile.am b/Makefile.am
--- a/Makefile.am	2011-05-26 19:18:54.000000000 +0200
+++ b/Makefile.am	2011-06-04 20:56:50.863488131 +0200
@@ -567,79 +567,13 @@
 	src/readahead-common.h \
 	src/ask-password-api.h
 
-MANPAGES = \
-	man/systemd.1 \
-	man/systemctl.1 \
-	man/systemadm.1 \
-	man/systemd-cgls.1 \
-	man/systemd-nspawn.1 \
-	man/systemd-tmpfiles.8 \
-	man/systemd-notify.1 \
-	man/sd_notify.3 \
-	man/sd_readahead.3 \
-	man/sd_booted.3 \
-	man/sd_listen_fds.3 \
-	man/sd_is_fifo.3 \
-	man/systemd.unit.5 \
-	man/systemd.service.5 \
-	man/systemd.socket.5 \
-	man/systemd.mount.5 \
-	man/systemd.automount.5 \
-	man/systemd.swap.5 \
-	man/systemd.timer.5 \
-	man/systemd.path.5 \
-	man/systemd.target.5 \
-	man/systemd.device.5 \
-	man/systemd.snapshot.5 \
-	man/systemd.exec.5 \
-	man/daemon.7 \
-	man/sd-daemon.7 \
-	man/sd-readahead.7 \
-	man/runlevel.8 \
-	man/telinit.8 \
-	man/halt.8 \
-	man/shutdown.8 \
-	man/pam_systemd.8 \
-	man/systemd.conf.5 \
-	man/tmpfiles.d.5 \
-	man/hostname.5 \
-	man/machine-id.5 \
-	man/vconsole.conf.5 \
-	man/locale.conf.5 \
-	man/os-release.5 \
-        man/machine-info.5 \
-	man/modules-load.d.5 \
-	man/sysctl.d.5 \
-        man/systemd-ask-password.1
+MANPAGES = #
 
-if ENABLE_BINFMT
-MANPAGES += \
-	man/binfmt.d.5
-endif
+MANPAGES_ALIAS = #
 
-MANPAGES_ALIAS = \
-	man/reboot.8 \
-	man/poweroff.8 \
-	man/sd_is_socket.3 \
-	man/sd_is_socket_unix.3 \
-	man/sd_is_socket_inet.3 \
-	man/sd_notifyf.3 \
-	man/init.1
-
-man/reboot.8: man/halt.8
-man/poweroff.8: man/halt.8
-man/sd_is_socket.3: man/sd_is_fifo.3
-man/sd_is_socket_unix.3: man/sd_is_fifo.3
-man/sd_is_socket_inet.3: man/sd_is_fifo.3
-man/sd_notifyf.3: man/sd_notify.3
-man/init.1: man/systemd.1
-
-dist_man_MANS = \
-	$(MANPAGES) \
-	$(MANPAGES_ALIAS)
+dist_man_MANS = #
 
-nodist_man_MANS = \
-	man/systemd.special.7
+nodist_man_MANS = #
 
 XML_FILES = \
 	${patsubst %.1,%.xml,${patsubst %.3,%.xml,${patsubst %.5,%.xml,${patsubst %.7,%.xml,${patsubst %.8,%.xml,$(MANPAGES)}}}}}
