diff -Nur a/tests/Makefile.am b/tests/Makefile.am
--- a/tests/Makefile.am	2012-08-18 08:02:47.000000000 +0200
+++ b/tests/Makefile.am	2012-08-26 14:57:03.065516651 +0200
@@ -24,7 +24,6 @@
 
 root_tests =					\
   chown/basic					\
-  cp/cp-a-selinux				\
   cp/preserve-gid				\
   cp/special-bits				\
   cp/cp-mv-enotsup-xattr			\
@@ -199,7 +198,6 @@
   misc/groups-version				\
   misc/head-c					\
   misc/head-pos					\
-  misc/id-context				\
   misc/id-groups				\
   misc/id-setgid				\
   misc/md5sum					\
@@ -252,7 +250,6 @@
   misc/sort-merge				\
   misc/sort-merge-fdlimit			\
   misc/sort-month				\
-  misc/sort-exit-early				\
   misc/sort-rand				\
   misc/sort-spinlock-abuse			\
   misc/sort-stale-thread-mem			\
@@ -421,12 +418,10 @@
   du/trailing-slash				\
   du/two-args					\
   id/gnu-zero-uids				\
-  id/no-context					\
   install/basic-1				\
   install/create-leading			\
   install/d-slashdot				\
   install/install-C				\
-  install/install-C-selinux			\
   install/strip-program				\
   install/trap					\
   ln/backup-1					\
