diff -Nur a/tests/Makefile.am b/tests/Makefile.am
--- a/tests/Makefile.am	2012-03-24 19:22:13.000000000 +0100
+++ b/tests/Makefile.am	2012-04-28 15:42:22.861910633 +0200
@@ -24,7 +24,6 @@
 
 root_tests =					\
   chown/basic					\
-  cp/cp-a-selinux				\
   cp/preserve-gid				\
   cp/special-bits				\
   cp/cp-mv-enotsup-xattr			\
@@ -195,7 +194,6 @@
   misc/groups-version				\
   misc/head-c					\
   misc/head-pos					\
-  misc/id-context				\
   misc/id-groups				\
   misc/md5sum					\
   misc/md5sum-bsd				\
@@ -408,12 +406,10 @@
   du/trailing-slash				\
   du/two-args					\
   id/gnu-zero-uids				\
-  id/no-context					\
   install/basic-1				\
   install/create-leading			\
   install/d-slashdot				\
   install/install-C				\
-  install/install-C-selinux			\
   install/strip-program				\
   install/trap					\
   ln/backup-1					\
