diff -Nur coreutils-8.23-orig/tests/local.mk coreutils-8.23/tests/local.mk
--- coreutils-8.23-orig/tests/local.mk	2014-07-14 00:09:52.000000000 +0200
+++ coreutils-8.23/tests/local.mk	2015-01-23 21:44:03.355661943 +0100
@@ -107,7 +107,6 @@
 
 all_root_tests =				\
   tests/chown/basic.sh				\
-  tests/cp/cp-a-selinux.sh			\
   tests/cp/preserve-gid.sh			\
   tests/cp/special-bits.sh			\
   tests/cp/cp-mv-enotsup-xattr.sh		\
@@ -121,8 +120,6 @@
   tests/install/install-C-root.sh		\
   tests/ls/capability.sh			\
   tests/ls/nameless-uid.sh			\
-  tests/misc/chcon.sh				\
-  tests/misc/chroot-credentials.sh		\
   tests/misc/selinux.sh				\
   tests/misc/truncate-owned-by-other.sh		\
   tests/mkdir/writable-under-readonly.sh	\
@@ -517,8 +514,6 @@
   tests/du/trailing-slash.sh			\
   tests/du/two-args.sh				\
   tests/id/gnu-zero-uids.sh			\
-  tests/id/no-context.sh			\
-  tests/id/context.sh				\
   tests/id/uid.sh				\
   tests/id/setgid.sh				\
   tests/id/zero.sh				\
@@ -527,7 +522,6 @@
   tests/install/create-leading.sh		\
   tests/install/d-slashdot.sh			\
   tests/install/install-C.sh			\
-  tests/install/install-C-selinux.sh		\
   tests/install/strip-program.sh		\
   tests/install/trap.sh				\
   tests/ln/backup-1.sh				\
@@ -579,7 +573,6 @@
   tests/mkdir/parents.sh			\
   tests/mkdir/perm.sh				\
   tests/mkdir/selinux.sh			\
-  tests/mkdir/restorecon.sh			\
   tests/mkdir/special-1.sh			\
   tests/mkdir/t-slash.sh			\
   tests/mkdir/smack-no-root.sh			\
