From 2d80bda39073a35af4b904d27fa1511cd309b26f Mon Sep 17 00:00:00 2001
From: Nathan Sidwell <nathan@codesourcery.com>
Date: Fri, 6 Jan 2012 20:14:44 +0000
Subject: [PATCH 01/17] Add ARM dependency of libmemusage.so on
 libc_nonshared.a.

---
 sysdeps/arm/Makefile |    5 +++++
 1 file changed, 5 insertions(+), 0 deletions(-)

diff --git a/ports/sysdeps/arm/Makefile b/ports/sysdeps/arm/Makefile
index 5651161..1a88430 100644
--- a/ports/sysdeps/arm/Makefile
+++ b/ports/sysdeps/arm/Makefile
@@ -7,3 +7,8 @@ endif
 ifeq ($(subdir),csu)
 gen-as-const-headers += tlsdesc.sym
 endif
+
+# to pull in __aeabi_read_tp, needed for tls
+ifeq ($(subdir),malloc)
+$(objpfx)libmemusage.so: $(common-objpfx)libc_nonshared.a
+endif
-- 
1.7.5.4

