From 4252a54980c0996a9b46ad2043aefe0910bc5da7 Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Tue, 5 Mar 2013 01:20:22 +0100
Subject: [PATCH 02/19] Mount /tmp as a tmpfs into the build environment.

---
 python/pakfire/builder.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/python/pakfire/builder.py b/python/pakfire/builder.py
index 24027d9..1cf1e8e 100644
--- a/python/pakfire/builder.py
+++ b/python/pakfire/builder.py
@@ -572,6 +572,7 @@ class BuildEnviron(object):
 			("pakfire_tmpfs", "/dev",      "tmpfs", "mode=755,nosuid"),
 			("/dev/pts",      "/dev/pts",  "bind",  "bind"),
 			("pakfire_tmpfs", "/run",      "tmpfs", "mode=755,nosuid,nodev"),
+			("pakfire_tmpfs", "/tmp",      "tmpfs", "mode=755,nosuid,nodev"),
 		]
 
 		# If selinux is enabled.
-- 
1.8.1.4

