From 7ac1cb68d8dfc00d7189abb63f8af710ed276385 Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Tue, 2 Oct 2012 22:07:50 +0200
Subject: [PATCH 4/4] macros: Add templates for python packages.

---
 macros/templates.macro | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/macros/templates.macro b/macros/templates.macro
index 794619b..a2388b9 100644
--- a/macros/templates.macro
+++ b/macros/templates.macro
@@ -87,4 +87,24 @@ packages
 			%{datadir}/doc
 		end
 	end
+
+	template PYTHON
+		summary = Python bindings from the %{name} package.
+		description = %{summary}
+
+		files
+			%{python_sitearch}
+			%{python_sitelib}
+		end
+	end
+
+	template PYTHON3
+		summary = Python 3 bindings from the %{name} package.
+		description = %{summary}
+
+		files
+			%{python3_sitearch}
+			%{python3_sitelib}
+		end
+	end
 end
-- 
1.7.11.4

