From a4104146edcbc0d646f5e68c366735e514ad295e Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Thu, 24 Jan 2013 02:37:27 +0100
Subject: [PATCH 05/17] pakfire.packages.solv: Remove duplicate method "uuid".

---
 python/pakfire/packages/solv.py | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/python/pakfire/packages/solv.py b/python/pakfire/packages/solv.py
index 85bf487..af2c947 100644
--- a/python/pakfire/packages/solv.py
+++ b/python/pakfire/packages/solv.py
@@ -135,10 +135,6 @@ class SolvPackage(base.Package):
 		return vendor
 
 	@property
-	def uuid(self):
-		return self.solvable.get_uuid()
-
-	@property
 	def build_host(self):
 		return self.solvable.get_buildhost()
 
-- 
1.8.1.2

