From acf9ff3a740f37550287b4e263e6d7e76cef3e36 Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Sun, 22 Apr 2012 14:42:37 +0200
Subject: [PATCH 05/16] Allow to install multiple non-default kernels.

---
 python/pakfire/constants.py |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/python/pakfire/constants.py b/python/pakfire/constants.py
index 0899dd0..44d1d02 100644
--- a/python/pakfire/constants.py
+++ b/python/pakfire/constants.py
@@ -184,7 +184,14 @@ end
 PACKAGE_INFO_DESCRIPTION_LINE = PACKAGE_INFO_DEPENDENCY_LINE = "\t\t%s"
 
 # XXX make this configurable in pakfire.conf
-PAKFIRE_MULTIINSTALL = ["kernel", "kernel-PAE",]
+PAKFIRE_MULTIINSTALL = [
+	"kernel",
+	"kernel-kirkwood",
+	"kernel-legacy",
+	"kernel-omap",
+	"kernel-versatile",
+	"kernel-devel",
+]
 
 SCRIPTLET_INTERPRETER = "/bin/sh"
 SCRIPTLET_TIMEOUT = 60 * 15
-- 
1.7.10.4

