From 337a733a278a81c1ad44f9d1fed65ec6581dfa44 Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Sun, 10 Mar 2013 01:22:19 +0100
Subject: [PATCH 06/19] cgroup: Don't be verbose about killing processes.

---
 python/pakfire/cgroup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python/pakfire/cgroup.py b/python/pakfire/cgroup.py
index 6c85937..207f270 100644
--- a/python/pakfire/cgroup.py
+++ b/python/pakfire/cgroup.py
@@ -316,7 +316,7 @@ class CGroup(object):
 
 			if sig:
 				# Send sig to all processes in the cgroup.
-				log.info("Sending signal %s to all processes in '%s'." % (sig, self.name))
+				log.debug("Sending signal %s to all processes in '%s'." % (sig, self.name))
 				self.kill(sig=sig, recursive=True)
 
 			# Sleep for 200ms.
-- 
1.8.1.4

