From b8f51d988734a75e8e41496a5d272057d18ecfda Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Sun, 22 Apr 2012 14:47:43 +0200
Subject: [PATCH 07/16] Select packages from the database by their UUIDs.

---
 python/pakfire/repository/database.py |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/python/pakfire/repository/database.py b/python/pakfire/repository/database.py
index 11321ae..6f8f397 100644
--- a/python/pakfire/repository/database.py
+++ b/python/pakfire/repository/database.py
@@ -334,9 +334,9 @@ class DatabaseLocal(Database):
 
 		# Get the ID of the package in the database.
 		c = self.cursor()
-		#c.execute("SELECT id FROM packages WHERE uuid = ? LIMIT 1", (pkg.uuid,))
-		c.execute("SELECT id FROM packages WHERE name = ? AND epoch = ? AND version = ?"
-			" AND release = ? LIMIT 1", (pkg.name, pkg.epoch, pkg.version, pkg.release,))
+		c.execute("SELECT id FROM packages WHERE uuid = ? LIMIT 1", (pkg.uuid,))
+		#c.execute("SELECT id FROM packages WHERE name = ? AND epoch = ? AND version = ?"
+		#	" AND release = ? LIMIT 1", (pkg.name, pkg.epoch, pkg.version, pkg.release,))
 
 		id = None
 		for row in c:
-- 
1.7.10.4

