From c844646a49e8d30db91f75d115b28886b4775a94 Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Thu, 24 Jan 2013 02:43:54 +0100
Subject: [PATCH 07/17] Fix wrong variable name.

---
 python/pakfire/packages/file.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/python/pakfire/packages/file.py b/python/pakfire/packages/file.py
index 990f6f7..039103e 100644
--- a/python/pakfire/packages/file.py
+++ b/python/pakfire/packages/file.py
@@ -132,7 +132,8 @@ class FilePackage(base.Package):
 			payload_archive = tar.InnerTarFile.open(fileobj=payload)
 
 		else:
-			raise Exception, "Unhandled payload compression type: %s" % payload_compression
+			raise Exception, "Unhandled payload compression type: %s" % \
+				self.payload_compression
 
 		return payload_archive
 
-- 
1.8.1.2

