From 61918bfab20eaba61bb8ce7e8397a28806d029d2 Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Wed, 3 Oct 2012 14:56:41 +0200
Subject: [PATCH 8/8] py-compile: Make script more robust for errors.

---
 tools/py-compile | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/tools/py-compile b/tools/py-compile
index 5ccd8f9..2749042 100755
--- a/tools/py-compile
+++ b/tools/py-compile
@@ -33,14 +33,18 @@ tempfile=$(mktemp)
 trap "rm -f ${tempfile}" EXIT
 
 cat > ${tempfile} <<EOF
+import os
 import py_compile
 import sys
 
 for file in sys.argv[1:]:
+	if not os.path.exists(file):
+		continue
+
 	py_compile.compile(file, doraise=0)
 EOF
 
-filelist=$(find ${paths} -type f -a -name "*.py")
+filelist=$(find ${paths} -type f -a -name "*.py" 2>/dev/null)
 
 # Compile with optimization.
 ${python_interpreter} -O ${tempfile} ${filelist}
@@ -49,7 +53,7 @@ ${python_interpreter} -O ${tempfile} ${filelist}
 ${python_interpreter} ${tempfile} ${filelist}
 
 # Hardlink identical files.
-for pyc in $(find ${paths} -type f -a -name "*.pyc"); do
+for pyc in $(find ${paths} -type f -a -name "*.pyc" 2>/dev/null); do
 	pyo="${pyc/.pyc/.pyo}"
 
 	if cmp -s "${pyc}" "${pyo}"; then
-- 
1.7.11.4

