From 9accbba591014c8b2b0cb1c0b7c4fbf914caba8e Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Thu, 24 Jan 2013 02:44:53 +0100
Subject: [PATCH 08/17] FilePackage: Initialize _type.

---
 python/pakfire/packages/file.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/python/pakfire/packages/file.py b/python/pakfire/packages/file.py
index 039103e..7cf3630 100644
--- a/python/pakfire/packages/file.py
+++ b/python/pakfire/packages/file.py
@@ -46,6 +46,8 @@ class FilePackage(base.Package):
 		This class is a wrapper that reads package data from the (outer)
 		tarball and should never be used solely.
 	"""
+	_type = None
+
 	def __init__(self, pakfire, repo, filename):
 		base.Package.__init__(self, pakfire, repo)
 		self.filename = os.path.abspath(filename)
-- 
1.8.1.2

