From 0757c740d0551dc3a0013d55595579a71285c124 Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Sun, 7 Apr 2013 14:15:01 +0200
Subject: [PATCH 10/19] Fix typo.

---
 python/pakfire/daemon.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python/pakfire/daemon.py b/python/pakfire/daemon.py
index 267b330..67fdd69 100644
--- a/python/pakfire/daemon.py
+++ b/python/pakfire/daemon.py
@@ -101,7 +101,7 @@ class PakfireDaemon(object):
 		self.__running = False
 
 	def restart_keepalive(self, wait=None):
-		log.critial(_("Restarting keepalive process"))
+		log.critical(_("Restarting keepalive process"))
 
 		# Send SIGTERM to really end the process.
 		self.keepalive.terminate()
-- 
1.8.1.4

