From 90c9f2492736ab3a1d9b202da10803a248b8c0bc Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Wed, 8 May 2013 20:31:27 +0000
Subject: [PATCH 11/19] system: Fix reading CPU info.

---
 python/pakfire/system.py | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/python/pakfire/system.py b/python/pakfire/system.py
index 1f0f6de..a917461 100644
--- a/python/pakfire/system.py
+++ b/python/pakfire/system.py
@@ -130,17 +130,19 @@ class System(object):
 	def cpu_model(self):
 		cpuinfo = self.parse_cpuinfo()
 
-		ret = None
-		if self.arch.startswith("arm"):
+		ret = cpuinfo.get("model name", None)
+
+		# Some ARM platforms do not provide "model name", so we
+		# try an other way.
+		if ret is None:
 			try:
 				ret = "%(Hardware)s - %(Processor)s" % cpuinfo
 			except KeyError:
 				pass
-		else:
-			ret = cpuinfo.get("model name", None)
 
 		# Remove too many spaces.
-		ret = " ".join(ret.split())
+		if ret:
+			ret = " ".join(ret.split())
 
 		return ret or _("Could not be determined")
 
-- 
1.8.1.4

