From 493336ff5c795b6d33d78659a69cc16165e7d0d8 Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Thu, 13 Jun 2013 19:18:23 +0200
Subject: [PATCH 14/19] Better formatting for progress bars.

---
 python/pakfire/util.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python/pakfire/util.py b/python/pakfire/util.py
index 8d91709..73c62d0 100644
--- a/python/pakfire/util.py
+++ b/python/pakfire/util.py
@@ -86,7 +86,7 @@ def make_progress(message, maxval, eta=True, speed=False):
 
 	widgets = [
 		"  ",
-		"%s" % message,
+		"%-64s" % message,
 		" ",
 		progressbar.Bar(left="[", right="]"),
 		"  ",
-- 
1.8.1.4

