From 42d6af74306ecb1627eb3c5b6a0163fc6308b46d Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Sat, 30 Jun 2012 13:32:23 +0200
Subject: [PATCH 16/16] template: Fix packaging cmake files.

---
 macros/templates.macro |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/macros/templates.macro b/macros/templates.macro
index 76dcf69..794619b 100644
--- a/macros/templates.macro
+++ b/macros/templates.macro
@@ -52,7 +52,7 @@ packages
 			%{libdir}/pkgconfig
 			%{datadir}/aclocal
 			%{libdir}/*.so
-			%{datadir}/*/cmake
+			%{datadir}/cmake
 			%{mandir}/man2
 			%{mandir}/man3
 			%{datadir}/pkgconfig
-- 
1.7.10.4

