From 326e179e6cb035b9f789e2b62e193369b3f30d00 Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Tue, 19 Feb 2013 23:27:14 +0100
Subject: [PATCH 17/17] Template macros: Remove trailing slash.

---
 macros/templates.macro | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/macros/templates.macro b/macros/templates.macro
index 693fa12..c4cda8c 100644
--- a/macros/templates.macro
+++ b/macros/templates.macro
@@ -55,7 +55,7 @@ packages
 			%{datadir}/cmake
 			%{mandir}/man2
 			%{mandir}/man3
-			%{datadir}/gir*/
+			%{datadir}/gir*
 			%{datadir}/pkgconfig
 			%{datadir}/vala
 		end
-- 
1.8.1.2

