From cf1301293b1eea70faa2dc47f4c788aa7e52960c Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Mon, 24 Jun 2013 21:34:45 +0000
Subject: [PATCH 18/19] Fix installing unit files.

---
 Makeconfig | 3 +++
 Makefile   | 5 ++++-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/Makeconfig b/Makeconfig
index b9f7898..ec6a2f2 100644
--- a/Makeconfig
+++ b/Makeconfig
@@ -50,3 +50,6 @@ TOP := $(dir $(lastword $(MAKEFILE_LIST)))
 # A list of all files that contain translations and need to
 # be indexed.
 TRANS_FILES  = $(PYTHON_FILES) python/src/*.c
+
+# systemd unit files
+UNIT_FILES = $(wildcard systemd/*.service)
diff --git a/Makefile b/Makefile
index 00ea0fa..2a1497b 100644
--- a/Makefile
+++ b/Makefile
@@ -40,7 +40,10 @@ install: build
 
 	# Install systemd file.
 	-mkdir -pv $(DESTDIR)/usr/lib/systemd/system
-	cp -vf systemd/*.systemd $(DESTDIR)/usr/lib/systemd/system
+	for file in $(UNIT_FILES); do \
+		install -v -m 644 $${file} \
+			$(DESTDIR)/usr/lib/systemd/system || exit 1; \
+	done
 
 .PHONY: check
 check: all
-- 
1.8.1.4

