From 123f594fe738d6eb0a9238d094bbfbce7d4355b3 Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Sat, 10 Mar 2012 11:45:20 +0100
Subject: [PATCH] QA: Don't use ls to list all directories in libs-location check.

---
 tools/quality-agent.d/003-libs-location |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/tools/quality-agent.d/003-libs-location b/tools/quality-agent.d/003-libs-location
index ce5ba72..4c2e8ba 100755
--- a/tools/quality-agent.d/003-libs-location
+++ b/tools/quality-agent.d/003-libs-location
@@ -19,7 +19,7 @@ gcc_libdir=$(gcc -print-libgcc-file-name)
 for i in 1 2; do
 	gcc_libdir=$(dirname ${gcc_libdir})
 done
-gcc_libdir=$(ls ${BUILDROOT}${gcc_libdir}/* 2>/dev/null | tail -n1)
+gcc_libdir=$(find ${BUILDROOT}${gcc_libdir}/* -type d -maxdepth 1 2>/dev/null | sort | head -n1)
 
 function check() {
 	local failed=0
-- 
1.7.3.4

