From 7cefdc0a2cb917e16c95d522fb1e3d4ed600faca Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Wed, 7 Mar 2012 17:11:37 +0100
Subject: [PATCH] Fix definition of sharedstatedir.

---
 macros/arch.macro |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/macros/arch.macro b/macros/arch.macro
index e756b8e..536e9d2 100644
--- a/macros/arch.macro
+++ b/macros/arch.macro
@@ -8,7 +8,7 @@ sbindir           = %{exec_prefix}/sbin
 libexecdir        = %{exec_prefix}/lib
 datadir           = %{prefix}/share
 sysconfdir        = /etc
-sharedstatedir    = %{prefix}/com
+sharedstatedir    = %{localstatedir}/lib
 localstatedir     = /var
 lib               = lib
 libdir            = %{exec_prefix}/%{lib}
-- 
1.7.3.4

