From cd37f1637dc038f2ebdfc514882e4b600d85a606 Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Sat, 10 Mar 2012 13:25:41 +0100
Subject: [PATCH] QA: Check for nested $BUILDROOT.

---
 tools/quality-agent.d/095-directory-layout |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/tools/quality-agent.d/095-directory-layout b/tools/quality-agent.d/095-directory-layout
index 3f946a1..d605fe9 100755
--- a/tools/quality-agent.d/095-directory-layout
+++ b/tools/quality-agent.d/095-directory-layout
@@ -4,7 +4,7 @@
 
 DESC="The filelayout should comply to the FHS."
 
-DIRS="/etc/init.d /etc/rc.d /lib/pkgconfig /usr/etc /usr/libexec /usr/local /usr/man /usr/usr /usr/var"
+DIRS="/etc/init.d /etc/rc.d /lib/pkgconfig /usr/etc /usr/libexec /usr/local /usr/man /usr/usr /usr/var ${BUILDROOT}"
 
 function check() {
 	# Do nothing, if directory check was disabled.
-- 
1.7.3.4

