commit da1de278eca7d6663cce89f7c1ad2a312fda8abb
Author: Michael Tremer <michael.tremer@ipfire.org>
Date:   Sun Mar 11 15:03:21 2012 +0100

    QA: libs-location: Don't search in the deep.

diff --git a/tools/quality-agent.d/003-libs-location b/tools/quality-agent.d/003-libs-location
index 4c2e8ba..479f170 100755
--- a/tools/quality-agent.d/003-libs-location
+++ b/tools/quality-agent.d/003-libs-location
@@ -25,7 +25,7 @@ function check() {
 	local failed=0
 	local found
 
-	for lib in $(find ${BUILDROOT}/{,usr/}${libdir} -type f -name "lib*.so.*" 2>/dev/null); do
+	for lib in $(find ${BUILDROOT}/{,usr/}${libdir} -maxdepth 1 -type f -name "lib*.so.*" 2>/dev/null); do
 		lib=${lib##*/}
 		lib=${lib%%.so*}
 
