diff -Nur a/libs/context/src/asm/fcontext_arm_aapcs_elf_gas.S b/libs/context/src/asm/fcontext_arm_aapcs_elf_gas.S
--- a/libs/context/src/asm/fcontext_arm_aapcs_elf_gas.S	2012-07-03 21:12:31.000000000 +0200
+++ b/libs/context/src/asm/fcontext_arm_aapcs_elf_gas.S	2012-10-27 19:25:08.016850683 +0200
@@ -45,6 +45,10 @@
  *                                                                 *
  * *****************************************************************/
 
+#if defined(__linux__) && defined(__ELF__)
+.section .note.GNU-stack,"",%progbits
+#endif
+
 .text
 .globl jump_fcontext
 .align 2
diff -Nur a/libs/context/src/asm/fcontext_i386_sysv_elf_gas.S b/libs/context/src/asm/fcontext_i386_sysv_elf_gas.S
--- a/libs/context/src/asm/fcontext_i386_sysv_elf_gas.S	2012-07-03 21:12:31.000000000 +0200
+++ b/libs/context/src/asm/fcontext_i386_sysv_elf_gas.S	2012-10-27 19:24:34.166680913 +0200
@@ -31,6 +31,10 @@
  *                                                                  *
  * *****************************************************************/
 
+#if defined(__linux__) && defined(__ELF__)
+.section .note.GNU-stack,"",%progbits
+#endif
+
 .text
 .globl jump_fcontext
 .align 2
