From c3dbb53b6c206a0fdfacad9915a2df0c3613089a Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Sun, 22 Jul 2012 11:36:11 +0000
Subject: [PATCH] sqlite3: Add more logging if database initialization fails.

---
 output/sqlite3/ulogd_output_SQLITE3.c |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/output/sqlite3/ulogd_output_SQLITE3.c b/output/sqlite3/ulogd_output_SQLITE3.c
index dffdda3..6f371c3 100644
--- a/output/sqlite3/ulogd_output_SQLITE3.c
+++ b/output/sqlite3/ulogd_output_SQLITE3.c
@@ -322,8 +322,10 @@ sqlite3_init_db(struct ulogd_pluginstance *pi)
 	sqlite3_stmt *schema_stmt;
 	int col, num_cols;
 
-	if (priv->dbh == NULL)
+	if (priv->dbh == NULL) {
+		ulogd_log(ULOGD_ERROR, "SQLITE3: No database handle.\n");
 		return -1;
+	}
 
 	num_cols = db_count_cols(pi, &schema_stmt);
 	if (num_cols <= 0) {
@@ -397,8 +399,10 @@ sqlite3_start(struct ulogd_pluginstance *pi)
 	sqlite3_busy_timeout(priv->dbh, SQLITE3_BUSY_TIMEOUT);
 
 	/* read the fieldnames to know which values to insert */
-	if (sqlite3_init_db(pi) < 0)
+	if (sqlite3_init_db(pi) < 0) {
+		ulogd_log(ULOGD_ERROR, "SQLITE3: Could not read database fieldnames.\n");
 		return -1;
+	}
 
 	/* initialize our buffer size and counter */
 	priv->buffer_size = buffer_ce(pi);
-- 
1.7.8.2

