--- xen-4.0.1/tools/Rules.mk      2010-09-26 19:28:02.748059404 +0000
+++ xen-4.0.1/tools/Rules.mk      2010-09-26 19:25:30.769293405 +0000
@@ -15,6 +15,7 @@ XEN_XENSTORE       = $(XEN_ROOT)/tools/x
 XEN_LIBXENSTAT     = $(XEN_ROOT)/tools/xenstat/libxenstat/src

 CFLAGS_include = -I$(XEN_INCLUDE)
+CFLAGS-y               += -fno-pie -fno-stack-protector

 CFLAGS_libxenctrl = -I$(XEN_LIBXC) $(CFLAGS_include)
 LDFLAGS_libxenctrl = -L$(XEN_LIBXC) -lxenctrl

