diff -Nur a/makeinfo/tests/accentenc b/makeinfo/tests/accentenc
--- a/makeinfo/tests/accentenc	2004-04-11 19:56:47.000000000 +0200
+++ b/makeinfo/tests/accentenc	2011-07-27 20:24:36.603237061 +0200
@@ -1,7 +1,8 @@
 #!/bin/sh
 # Test encoded accent info output.
 
-unset TEXINFO_OUTPUT
+unset TEXINFO_OUTPUT LANG LANGUAGE
+LC_ALL=POSIX; export LC_ALL
 : ${srcdir=.}
 input=`basename $0`.txi
 output=`basename $0`.info
