diff -Nur a/Makefile b/Makefile
--- a/Makefile	2010-11-19 13:26:11.000000000 +0100
+++ b/Makefile	2010-11-19 13:27:31.000000000 +0100
@@ -525,11 +525,11 @@
 man/init.1: man/systemd.1
 
 dist_man_MANS = \
-	$(MANPAGES) \
-	$(MANPAGES_ALIAS)
+#	$(MANPAGES) \
+#	$(MANPAGES_ALIAS)
 
 nodist_man_MANS = \
-	man/systemd.special.7
+#	man/systemd.special.7
 
 XML_FILES = \
 	${patsubst %.1,%.xml,${patsubst %.3,%.xml,${patsubst %.5,%.xml,${patsubst %.7,%.xml,${patsubst %.8,%.xml,$(MANPAGES)}}}}}
@@ -538,10 +538,10 @@
 	${patsubst %.1,%.xml.in,${patsubst %.3,%.xml.in,${patsubst %.5,%.xml.in,${patsubst %.7,%.xml.in,${patsubst %.8,%.xml.in,$(nodist_man_MANS)}}}}}
 
 dist_noinst_DATA = \
-	${XML_FILES:.xml=.html}
+#	${XML_FILES:.xml=.html}
 
 nodist_noinst_DATA = \
-	${XML_IN_FILES:.xml.in=.html}
+#	${XML_IN_FILES:.xml.in=.html}
 
 EXTRA_DIST += \
 	$(XML_FILES) \
