From 26cbf29c52a36b6ad9d1ccc16d8f7adccefeddca Mon Sep 17 00:00:00 2001
From: Kay Sievers <kay@vrfy.org>
Date: Thu, 07 Jun 2012 17:27:17 +0000
Subject: delete plymouth units; moved to plymouth

---
diff --git a/Makefile.am b/Makefile.am
index 6cc352a..f45fa27 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -425,25 +425,6 @@ systemgenerator_PROGRAMS += \
 	systemd-rc-local-generator
 endif
 
-if HAVE_PLYMOUTH
-dist_systemunit_DATA += \
-	units/plymouth-start.service \
-	units/plymouth-read-write.service \
-	units/plymouth-quit.service \
-	units/plymouth-quit-wait.service \
-	units/plymouth-reboot.service \
-	units/plymouth-kexec.service \
-	units/plymouth-poweroff.service \
-	units/plymouth-halt.service \
-	units/systemd-ask-password-plymouth.path
-
-nodist_systemunit_DATA += \
-	units/systemd-ask-password-plymouth.service
-
-EXTRA_DIST += \
-	units/systemd-ask-password-plymouth.service.in
-endif
-
 dist_doc_DATA = \
 	README \
 	NEWS \
@@ -3249,33 +3230,6 @@ systemd-install-data-hook:
 	( cd $(DESTDIR)$(dbussessionservicedir) && \
 		rm -f org.freedesktop.systemd1.service && \
 		$(LN_S) ../system-services/org.freedesktop.systemd1.service org.freedesktop.systemd1.service )
-if HAVE_PLYMOUTH
-	$(MKDIR_P) -m 0755 \
-		$(DESTDIR)$(systemunitdir)/reboot.target.wants \
-		$(DESTDIR)$(systemunitdir)/kexec.target.wants \
-		$(DESTDIR)$(systemunitdir)/poweroff.target.wants \
-		$(DESTDIR)$(systemunitdir)/halt.target.wants
-	( cd $(DESTDIR)$(systemunitdir)/sysinit.target.wants && \
-		rm -f plymouth-start.service plymouth-read-write.service && \
-		$(LN_S) ../plymouth-start.service plymouth-start.service && \
-		$(LN_S) ../plymouth-read-write.service plymouth-read-write.service )
-	( cd $(DESTDIR)$(systemunitdir)/multi-user.target.wants && \
-		rm -f plymouth-quit.service plymouth-quit-wait.service && \
-		$(LN_S) ../plymouth-quit.service plymouth-quit.service && \
-		$(LN_S) ../plymouth-quit-wait.service plymouth-quit-wait.service )
-	( cd $(DESTDIR)$(systemunitdir)/reboot.target.wants && \
-		rm -f plymouth-reboot.service && \
-		$(LN_S) ../plymouth-reboot.service plymouth-reboot.service )
-	( cd $(DESTDIR)$(systemunitdir)/kexec.target.wants && \
-		rm -f plymouth-kexec.service && \
-		$(LN_S) ../plymouth-kexec.service plymouth-kexec.service )
-	( cd $(DESTDIR)$(systemunitdir)/poweroff.target.wants && \
-		rm -f plymouth-poweroff.service && \
-		$(LN_S) ../plymouth-poweroff.service plymouth-poweroff.service )
-	( cd $(DESTDIR)$(systemunitdir)/halt.target.wants && \
-		rm -f plymouth-halt.service && \
-		$(LN_S) ../plymouth-halt.service plymouth-halt.service )
-endif
 if TARGET_MEEGO
 	$(MKDIR_P) -m 0755 $(DESTDIR)$(systemunitdir)/final.target.wants
 	( cd $(DESTDIR)$(systemunitdir)/multi-user.target.wants && \
diff --git a/README b/README
index d922204..d23a3d6 100644
--- a/README
+++ b/README
@@ -60,7 +60,6 @@ REQUIREMENTS:
 
         util-linux > v2.18 (requires fsck -l, agetty -s)
         sulogin (from sysvinit-tools, optional but recommended)
-        plymouth (optional)
         dracut (optional)
 
         When systemd-hostnamed is used it is strongly recommended to
diff --git a/configure.ac b/configure.ac
index 59892cb..70f3e96 100644
--- a/configure.ac
+++ b/configure.ac
@@ -519,20 +519,17 @@ SYSTEM_SYSVINIT_PATH=/etc/init.d
 SYSTEM_SYSVRCND_PATH=/etc/rc.d
 
 M4_DEFINES=
-have_plymouth=no
 
 case $with_distro in
         fedora)
                 SYSTEM_SYSVINIT_PATH=/etc/rc.d/init.d
                 AC_DEFINE(TARGET_FEDORA, [], [Target is Fedora/RHEL])
                 M4_DEFINES=-DTARGET_FEDORA=1
-                have_plymouth=yes
                 ;;
         opensuse|suse)
                 SYSTEM_SYSVRCND_PATH=/etc/init.d
                 AC_DEFINE(TARGET_SUSE, [], [Target is openSUSE/SLE])
                 M4_DEFINES=-DTARGET_SUSE=1
-                have_plymouth=yes
                 ;;
         debian)
                 SYSTEM_SYSVRCND_PATH=/etc
@@ -565,19 +562,16 @@ case $with_distro in
                 SYSTEM_SYSVINIT_PATH=/etc/rc.d
                 AC_DEFINE(TARGET_FRUGALWARE, [], [Target is Frugalware])
                 M4_DEFINES=-DTARGET_FRUGALWARE=1
-                have_plymouth=yes
                 ;;
         altlinux)
                 SYSTEM_SYSVINIT_PATH=/etc/rc.d/init.d
                 AC_DEFINE(TARGET_ALTLINUX, [], [Target is ALTLinux])
                 M4_DEFINES=-DTARGET_ALTLINUX=1
-                have_plymouth=yes
                 ;;
         mandriva)
                 SYSTEM_SYSVINIT_PATH=/etc/rc.d/init.d
                 AC_DEFINE(TARGET_MANDRIVA, [], [Target is Mandriva])
                 M4_DEFINES=-DTARGET_MANDRIVA=1
-                have_plymouth=yes
                 ;;
         meego)
                 SYSTEM_SYSVINIT_PATH=
@@ -594,7 +588,6 @@ case $with_distro in
                 SYSTEM_SYSVINIT_PATH=/etc/rc.d/init.d
                 AC_DEFINE(TARGET_MAGEIA, [], [Target is Mageia])
                 M4_DISTRO_FLAG=-DTARGET_MAGEIA=1
-                have_plymouth=yes
                 ;;
         other)
                 ;;
@@ -635,11 +628,6 @@ AC_ARG_WITH([tty-gid],
         [AC_DEFINE_UNQUOTED(TTY_GID, [$withval], [GID of the 'tty' group])],
         [])
 
-AC_ARG_ENABLE(plymouth, AS_HELP_STRING([--enable-plymouth], [enable plymouth support]))
-if test -n "$enable_plymouth"; then
-        have_plymouth="$enable_plymouth"
-fi
-
 AM_CONDITIONAL(TARGET_FEDORA, test x"$with_distro" = xfedora)
 AM_CONDITIONAL(TARGET_SUSE, test x"$with_distro" = xsuse)
 AM_CONDITIONAL(TARGET_DEBIAN, test x"$with_distro" = xdebian)
@@ -655,7 +643,6 @@ AM_CONDITIONAL(TARGET_MEEGO, test x"$with_distro" = xmeego)
 AM_CONDITIONAL(TARGET_ANGSTROM, test x"$with_distro" = xangstrom)
 AM_CONDITIONAL(TARGET_MAGEIA, test x"$with_distro" = xmageia)
 
-AM_CONDITIONAL(HAVE_PLYMOUTH, test "$have_plymouth" = "yes")
 AM_CONDITIONAL(HAVE_SYSV_COMPAT, test "$SYSTEM_SYSV_COMPAT" = "yes")
 
 AC_ARG_WITH([dbuspolicydir],
@@ -747,7 +734,6 @@ AC_MSG_RESULT([
         timedated:               ${have_timedated}
         localed:                 ${have_localed}
         coredump:                ${have_coredump}
-        plymouth:                ${have_plymouth}
         firmware path:           ${FIRMWARE_PATH}
         usb.ids:                 ${USB_DATABASE}
         pci.ids:                 ${PCI_DATABASE}
diff --git a/units/.gitignore b/units/.gitignore
index dc5e1d4..eb2516d 100644
--- a/units/.gitignore
+++ b/units/.gitignore
@@ -11,7 +11,6 @@ console-shell.service
 systemd-sysctl.service
 systemd-ask-password-console.service
 rescue.service
-systemd-ask-password-plymouth.service
 systemd-ask-password-wall.service
 quotacheck.service
 fsck@.service
diff --git a/units/plymouth-halt.service b/units/plymouth-halt.service
deleted file mode 100644
index 5dca3cc..0000000
--- a/units/plymouth-halt.service
+++ b/dev/null
@@ -1,18 +0,0 @@
-#  This file is part of systemd.
-#
-#  systemd is free software; you can redistribute it and/or modify it
-#  under the terms of the GNU Lesser General Public License as published by
-#  the Free Software Foundation; either version 2.1 of the License, or
-#  (at your option) any later version.
-
-[Unit]
-Description=Show Plymouth Halt Screen
-After=getty@tty1.service prefdm.service plymouth-start.service
-Before=halt.service
-DefaultDependencies=no
-ConditionKernelCommandLine=!plymouth.enable=0
-
-[Service]
-ExecStart=/sbin/plymouthd --mode=shutdown
-ExecStartPost=-/bin/plymouth --show-splash
-Type=forking
diff --git a/units/plymouth-kexec.service b/units/plymouth-kexec.service
deleted file mode 100644
index 8c36b75..0000000
--- a/units/plymouth-kexec.service
+++ b/dev/null
@@ -1,18 +0,0 @@
-#  This file is part of systemd.
-#
-#  systemd is free software; you can redistribute it and/or modify it
-#  under the terms of the GNU Lesser General Public License as published by
-#  the Free Software Foundation; either version 2.1 of the License, or
-#  (at your option) any later version.
-
-[Unit]
-Description=Show Plymouth Reboot with kexec Screen
-After=getty@tty1.service prefdm.service plymouth-start.service
-Before=kexec.service
-DefaultDependencies=no
-ConditionKernelCommandLine=!plymouth.enable=0
-
-[Service]
-ExecStart=/sbin/plymouthd --mode=shutdown
-ExecStartPost=-/bin/plymouth --show-splash
-Type=forking
diff --git a/units/plymouth-poweroff.service b/units/plymouth-poweroff.service
deleted file mode 100644
index e69033a..0000000
--- a/units/plymouth-poweroff.service
+++ b/dev/null
@@ -1,18 +0,0 @@
-#  This file is part of systemd.
-#
-#  systemd is free software; you can redistribute it and/or modify it
-#  under the terms of the GNU Lesser General Public License as published by
-#  the Free Software Foundation; either version 2.1 of the License, or
-#  (at your option) any later version.
-
-[Unit]
-Description=Show Plymouth Power Off Screen
-After=getty@tty1.service prefdm.service plymouth-start.service
-Before=poweroff.service
-DefaultDependencies=no
-ConditionKernelCommandLine=!plymouth.enable=0
-
-[Service]
-ExecStart=/sbin/plymouthd --mode=shutdown
-ExecStartPost=-/bin/plymouth --show-splash
-Type=forking
diff --git a/units/plymouth-quit-wait.service b/units/plymouth-quit-wait.service
deleted file mode 100644
index 3801c88..0000000
--- a/units/plymouth-quit-wait.service
+++ b/dev/null
@@ -1,15 +0,0 @@
-#  This file is part of systemd.
-#
-#  systemd is free software; you can redistribute it and/or modify it
-#  under the terms of the GNU Lesser General Public License as published by
-#  the Free Software Foundation; either version 2.1 of the License, or
-#  (at your option) any later version.
-
-[Unit]
-Description=Wait for Plymouth Boot Screen to Quit
-After=rc-local.service plymouth-start.service systemd-user-sessions.service
-
-[Service]
-ExecStart=-/bin/plymouth --wait
-Type=oneshot
-TimeoutSec=20
diff --git a/units/plymouth-quit.service b/units/plymouth-quit.service
deleted file mode 100644
index 8b4860f..0000000
--- a/units/plymouth-quit.service
+++ b/dev/null
@@ -1,15 +0,0 @@
-#  This file is part of systemd.
-#
-#  systemd is free software; you can redistribute it and/or modify it
-#  under the terms of the GNU Lesser General Public License as published by
-#  the Free Software Foundation; either version 2.1 of the License, or
-#  (at your option) any later version.
-
-[Unit]
-Description=Terminate Plymouth Boot Screen
-After=rc-local.service plymouth-start.service systemd-user-sessions.service
-
-[Service]
-ExecStart=-/bin/plymouth quit
-Type=oneshot
-TimeoutSec=20
diff --git a/units/plymouth-read-write.service b/units/plymouth-read-write.service
deleted file mode 100644
index e8ef447..0000000
--- a/units/plymouth-read-write.service
+++ b/dev/null
@@ -1,16 +0,0 @@
-#  This file is part of systemd.
-#
-#  systemd is free software; you can redistribute it and/or modify it
-#  under the terms of the GNU Lesser General Public License as published by
-#  the Free Software Foundation; either version 2.1 of the License, or
-#  (at your option) any later version.
-
-[Unit]
-Description=Tell Plymouth To Write Out Runtime Data
-DefaultDependencies=no
-After=local-fs.target
-Before=sysinit.target
-
-[Service]
-ExecStart=-/bin/plymouth update-root-fs --read-write
-Type=oneshot
diff --git a/units/plymouth-reboot.service b/units/plymouth-reboot.service
deleted file mode 100644
index 5c52bc6..0000000
--- a/units/plymouth-reboot.service
+++ b/dev/null
@@ -1,18 +0,0 @@
-#  This file is part of systemd.
-#
-#  systemd is free software; you can redistribute it and/or modify it
-#  under the terms of the GNU Lesser General Public License as published by
-#  the Free Software Foundation; either version 2.1 of the License, or
-#  (at your option) any later version.
-
-[Unit]
-Description=Show Plymouth Reboot Screen
-After=getty@tty1.service prefdm.service plymouth-start.service
-Before=reboot.service
-DefaultDependencies=no
-ConditionKernelCommandLine=!plymouth.enable=0
-
-[Service]
-ExecStart=/sbin/plymouthd --mode=shutdown
-ExecStartPost=-/bin/plymouth --show-splash
-Type=forking
diff --git a/units/plymouth-start.service b/units/plymouth-start.service
deleted file mode 100644
index 2e358aa..0000000
--- a/units/plymouth-start.service
+++ b/dev/null
@@ -1,22 +0,0 @@
-#  This file is part of systemd.
-#
-#  systemd is free software; you can redistribute it and/or modify it
-#  under the terms of the GNU Lesser General Public License as published by
-#  the Free Software Foundation; either version 2.1 of the License, or
-#  (at your option) any later version.
-
-[Unit]
-Description=Show Plymouth Boot Screen
-DefaultDependencies=no
-Wants=systemd-ask-password-plymouth.path
-After=systemd-vconsole-setup.service systemd-udev-settle.service
-Before=systemd-ask-password-plymouth.service
-
-# Dracut informs us with this flag file if plymouth is already running
-ConditionPathExists=!/run/plymouth/pid
-ConditionKernelCommandLine=!plymouth.enable=0
-
-[Service]
-ExecStart=/sbin/plymouthd --mode=boot --pid-file=/run/plymouth/pid
-ExecStartPost=-/bin/plymouth --show-splash
-Type=forking
diff --git a/units/systemd-ask-password-plymouth.path b/units/systemd-ask-password-plymouth.path
deleted file mode 100644
index 9320b4a..0000000
--- a/units/systemd-ask-password-plymouth.path
+++ b/dev/null
@@ -1,20 +0,0 @@
-#  This file is part of systemd.
-#
-#  systemd is free software; you can redistribute it and/or modify it
-#  under the terms of the GNU Lesser General Public License as published by
-#  the Free Software Foundation; either version 2.1 of the License, or
-#  (at your option) any later version.
-
-[Unit]
-Description=Forward Password Requests to Plymouth Directory Watch
-Documentation=http://www.freedesktop.org/wiki/Software/systemd/PasswordAgents
-DefaultDependencies=no
-Conflicts=shutdown.target
-After=plymouth-start.service
-Before=basic.target shutdown.target
-ConditionKernelCommandLine=!plymouth.enable=0
-ConditionPathExists=/run/plymouth/pid
-
-[Path]
-DirectoryNotEmpty=/run/systemd/ask-password
-MakeDirectory=yes
diff --git a/units/systemd-ask-password-plymouth.service.in b/units/systemd-ask-password-plymouth.service.in
deleted file mode 100644
index ea80bbd..0000000
--- a/units/systemd-ask-password-plymouth.service.in
+++ b/dev/null
@@ -1,19 +0,0 @@
-#  This file is part of systemd.
-#
-#  systemd is free software; you can redistribute it and/or modify it
-#  under the terms of the GNU Lesser General Public License as published by
-#  the Free Software Foundation; either version 2.1 of the License, or
-#  (at your option) any later version.
-
-[Unit]
-Description=Forward Password Requests to Plymouth
-Documentation=http://www.freedesktop.org/wiki/Software/systemd/PasswordAgents
-DefaultDependencies=no
-Conflicts=shutdown.target
-After=plymouth-start.service
-Before=shutdown.target
-ConditionKernelCommandLine=!plymouth.enable=0
-ConditionPathExists=/run/plymouth/pid
-
-[Service]
-ExecStart=@rootbindir@/systemd-tty-ask-password-agent --watch --plymouth
--
cgit v0.9.0.2-2-gbebe
