diff -Nur a/configure.ac b/configure.ac
--- a/configure.ac	2012-03-16 01:46:57.000000000 +0100
+++ b/configure.ac	2012-03-24 20:43:33.723789319 +0100
@@ -396,7 +396,7 @@
 
 AC_PATH_PROG([M4], [m4])
 
-AC_ARG_WITH(distro, AS_HELP_STRING([--with-distro=DISTRO],[Specify the distribution to target: One of fedora, suse, debian, ubuntu, arch, gentoo, slackware, altlinux, mandriva, meego, mageia, angstrom or other]))
+AC_ARG_WITH(distro, AS_HELP_STRING([--with-distro=DISTRO],[Specify the distribution to target: One of fedora, suse, debian, ubuntu, arch, gentoo, slackware, altlinux, mandriva, meego, mageia, angstrom, ipfire or other]))
 if test "z$with_distro" = "z"; then
         if test "$cross_compiling" = yes; then
                 AC_MSG_WARN([Target distribution cannot be reliably detected when cross-compiling. You should specify it with --with-distro (see $0 --help for recognized distros)])
@@ -492,6 +492,13 @@
                 M4_DISTRO_FLAG=-DTARGET_MAGEIA=1
                 have_plymouth=yes
                 ;;
+	ipfire)
+		SYSTEM_SYSVINIT_PATH=
+		SYSTEM_SYSVRCND_PATH=
+		AC_DEFINE(TARGET_IPFIRE, [], [Target is IPFire])
+		M4_DEFINES=-DTARGET_IPFIRE=1
+		have_plymouth=yes
+		;;
         other)
                 ;;
         *)
@@ -550,6 +557,7 @@
 AM_CONDITIONAL(TARGET_MEEGO, test x"$with_distro" = xmeego)
 AM_CONDITIONAL(TARGET_ANGSTROM, test x"$with_distro" = xangstrom)
 AM_CONDITIONAL(TARGET_MAGEIA, test x"$with_distro" = xmageia)
+AM_CONDITIONAL(TARGET_IPFIRE, test x"$with_distro" = xipfire)
 
 AM_CONDITIONAL(HAVE_PLYMOUTH, test "$have_plymouth" = "yes")
 AM_CONDITIONAL(HAVE_SYSV_COMPAT, test "$SYSTEM_SYSV_COMPAT" = "yes")
