--- openssl-1.1.0g/test/recipes/40-test_rehash.t~	2018-01-28 19:08:01.151912658 +0000
+++ openssl-1.1.0g/test/recipes/40-test_rehash.t	2018-01-28 19:09:19.408454430 +0000
@@ -23,7 +23,7 @@
 plan skip_all => "test_rehash is not available on this platform"
     unless run(app(["openssl", "rehash", "-help"]));
 
-plan tests => 5;
+plan tests => 3;
 
 indir "rehash.$$" => sub {
     prepare();
@@ -42,21 +42,6 @@
        'Testing rehash operations on empty directory');
 }, create => 1, cleanup => 1;
 
-indir "rehash.$$" => sub {
-    prepare();
-    chmod 0500, curdir();
-  SKIP: {
-      if (!ok(!open(FOO, ">unwritable.txt"),
-              "Testing that we aren't running as a privileged user, such as root")) {
-          close FOO;
-          skip "It's pointless to run the next test as root", 1;
-      }
-      isnt(run(app(["openssl", "rehash", curdir()])), 1,
-           'Testing rehash operations on readonly directory');
-    }
-    chmod 0700, curdir();       # make it writable again, so cleanup works
-}, create => 1, cleanup => 1;
-
 sub prepare {
     my @pemsourcefiles = sort glob(srctop_file('test', "*.pem"));
     my @destfiles = ();
