diff -Nur old/CMakeLists.txt new/CMakeLists.txt
--- old/CMakeLists.txt	2011-04-10 23:55:09.506257000 +0200
+++ new/CMakeLists.txt	2011-04-10 23:46:30.612257002 +0200
@@ -32,7 +32,7 @@
 INCLUDE( ${CMAKE_SOURCE_DIR}/VERSION.cmake )
 
 FIND_PACKAGE(EXPAT REQUIRED)
-FIND_PACKAGE(Check REQUIRED)
+#FIND_PACKAGE(Check REQUIRED)
 FIND_PACKAGE(ZLIB REQUIRED)
 
 IF ( FEDORA )
@@ -190,10 +190,10 @@
 
 ADD_SUBDIRECTORY(src)
 ADD_SUBDIRECTORY(ext)
-ADD_SUBDIRECTORY(tools)
-ADD_SUBDIRECTORY(tests)
-ADD_SUBDIRECTORY(examples)
-ADD_SUBDIRECTORY(doc)
+#ADD_SUBDIRECTORY(tools)
+#ADD_SUBDIRECTORY(tests)
+#ADD_SUBDIRECTORY(examples)
+#ADD_SUBDIRECTORY(doc)
 
 MESSAGE(STATUS "version: ${VERSION}")
 
diff -Nur old/ext/CMakeLists.txt new/ext/CMakeLists.txt
--- old/ext/CMakeLists.txt	2011-04-10 23:55:09.530257000 +0200
+++ new/ext/CMakeLists.txt	2011-04-10 18:51:42.217487001 +0200
@@ -8,7 +8,7 @@
     ${libsatsolverext_SRCS} pool_fileconflicts.c repo_rpmdb.c)
 ENDIF (NOT DEBIAN)
 
-ADD_LIBRARY(satsolverext STATIC ${libsatsolverext_SRCS})
+ADD_LIBRARY(satsolverext SHARED ${libsatsolverext_SRCS})
 
 SET(libsatsolverext_HEADERS
     pool_fileconflicts.h repo_content.h repo_deltainfoxml.h repo_helix.h repo_products.h
@@ -16,7 +16,7 @@
     repo_susetags.h repo_updateinfoxml.h repo_write.h repo_zyppdb.h
     tools_util.h repo_deb.h sat_xfopen.h)
 
-SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
+SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC -lexpat")
 
 INSTALL(FILES ${libsatsolverext_HEADERS} DESTINATION "${CMAKE_INSTALL_PREFIX}/include/satsolver")
 INSTALL(TARGETS satsolverext LIBRARY DESTINATION ${LIB_INSTALL_DIR} ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
Binärdateien old/.git/index and new/.git/index sind verschieden.
diff -Nur old/src/CMakeLists.txt new/src/CMakeLists.txt
--- old/src/CMakeLists.txt	2011-04-10 23:55:09.536257000 +0200
+++ new/src/CMakeLists.txt	2011-04-10 23:52:01.689257003 +0200
@@ -6,7 +6,7 @@
     transaction.c rules.c problems.c
     chksum.c md5.c sha1.c sha2.c satversion.c)
 
-ADD_LIBRARY(satsolver STATIC ${libsatsolver_SRCS})
+ADD_LIBRARY(satsolver SHARED ${libsatsolver_SRCS})
 
 SET(libsatsolver_HEADERS
     bitmap.h evr.h hash.h policy.h poolarch.h poolvendor.h pool.h
diff -Nur old/tools/CMakeLists.txt new/tools/CMakeLists.txt
--- old/tools/CMakeLists.txt	2011-04-10 23:55:10.679257000 +0200
+++ new/tools/CMakeLists.txt	2011-04-10 23:53:42.340257007 +0200
@@ -28,8 +28,8 @@
 ADD_EXECUTABLE(updateinfoxml2solv updateinfoxml2solv.c)
 TARGET_LINK_LIBRARIES(updateinfoxml2solv toolstuff satsolverext satsolver ${EXPAT_LIBRARY})
 
-ADD_EXECUTABLE(deltainfoxml2solv deltainfoxml2solv.c)
-TARGET_LINK_LIBRARIES(deltainfoxml2solv toolstuff satsolverext satsolver ${EXPAT_LIBRARY})
+#ADD_EXECUTABLE(deltainfoxml2solv deltainfoxml2solv.c)
+#TARGET_LINK_LIBRARIES(deltainfoxml2solv toolstuff satsolverext satsolver ${EXPAT_LIBRARY})
 
 ADD_EXECUTABLE(repomdxml2solv repomdxml2solv.c)
 TARGET_LINK_LIBRARIES(repomdxml2solv toolstuff satsolverext satsolver ${EXPAT_LIBRARY})
@@ -53,7 +53,6 @@
     helix2solv
     rpmmd2solv
     updateinfoxml2solv
-    deltainfoxml2solv
     repomdxml2solv
     installcheck
     DESTINATION ${BIN_INSTALL_DIR})
