diff -up isdn4k-utils-CVS-2006-07-20/capi20/capi20.h.orig isdn4k-utils-CVS-2006-07-20/capi20/capi20.h
--- isdn4k-utils-CVS-2006-07-20/capi20/capi20.h.orig	2007-09-27 21:45:43.000000000 +0200
+++ isdn4k-utils-CVS-2006-07-20/capi20/capi20.h	2007-09-27 23:45:40.000000000 +0200
@@ -20,6 +20,7 @@
 #ifndef __CAPI20_H__
 #define __CAPI20_H__
 
+#include <sys/types.h>
 #include <sys/time.h>
 
 #ifdef __cplusplus
